### PR TITLE
Complete copying of objects works

### LINKED ISSUE #1: Add copying of State objects
Check documentation for the assignment for the complete description.

This feature needs to completely copy a state object so the copy will be its own unique object in memory.

This will be complete if the hash of the object is unique. Test this by changing some values in own object and printing both out in the terminal.

### PR DESCRIPTION
**Pull request by Rajivrocks-Ltd**
Closes #1 
Added copying of objects so that the copy is it's own new unique object (new memory address).

### CODE REVIEW

File: assignment_1/src/State.java [hunk 1]
```diff
@@ -5,75 +5,96 @@
...
+    }
 
-	public boolean isLeaf() {
-		return true;
-	}
+    /*
+        Create a copy method that will be used for a new copy constructor that will use this method to assign the fields
+        from the object you want to copy (state.x) to your new object (this.x), fairly Straight forward.
+    */
+    public void copy(State state) {
```
Comment (StevenW1999) [inline assignment_1/src/State.java]: It is specified in the assignment that the method should return a new State object so the method should look like this:

```suggestion
    public State copy(State state) {
```

File: assignment_1/src/State.java [hunk 2]
```diff
@@ -5,75 +5,96 @@
...
+        this.board = state.board;
+        this.food = state.food;
+        this.agentX = state.agentX;
+        this.agentY = state.agentY;
+        this.score = state.score;
+        this.turn = state.turn;
```
Comment (StevenW1999) [inline assignment_1/src/State.java]: Naturally this method should return a new state with the copied variables.
```suggestion
        this.turn = state.turn;
        return new State(this.board, this.food, this.agentX, this.agentY, this.score, this.turn)
```

**Review by StevenW1999 — CHANGES_REQUESTED** [review]
Rest looks good!

File: assignment_1/src/State.java (hunk 2 above)
Comment (Rajivrocks-Ltd) [reply assignment_1/src/State.java]: Yep will do this!

**Comment by Rajivrocks-Ltd** [comment]
@StevenW1999 addressed your concurns

**Pull request merged**